PortalSwift/Sources/PortalSwift/Utils/Blockchains/PortalBlockchain.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  PortalBlockchain.swift
+//
+//
+//  Created by Blake Williams on 3/30/24.
+//
+
+import Foundation
+
+public class PortalBlockchain {
+  static let mainnetReferences: [PortalNamespace: String] = [
+    .eip155: "1",
+  ]
+  static let namespaceCurves: [PortalNamespace: PortalCurve] = [
+    .eip155: .SECP256K1,
+    .solana: .ED25519,
+  ]
+  static let namespaceSignerMethods: [PortalNamespace: [PortalRequestMethod]] = [
+    .eip155: [
+      .eth_sendTransaction,
+      .eth_sign,
+      .eth_signTransaction,
+      .eth_signTypedData_v3,
+      .eth_signTypedData_v4,
+      .personal_sign,
+    ],
+  ]
+
+  public let curve: PortalCurve
+  public let isMainnet: Bool
+  public let namespace: PortalNamespace
+  public let reference: String?
+  private let signerMethods: [PortalRequestMethod]
+
+  init(fromChainId: String) throws {
+    let chainIdParts = fromChainId.split(separator: ":").map(String.init)
+    let namespaceString = chainIdParts[0]
+    let reference = chainIdParts[1]
+    guard !reference.isEmpty else {
+      throw PortalBlockchainError.invalidChainId(fromChainId)
+    }
+    guard let namespace = PortalNamespace(rawValue: namespaceString) else {
+      throw PortalBlockchainError.noSupportedNamespaceForChainId(fromChainId)
+    }
+    guard let curve = PortalBlockchain.namespaceCurves[namespace] else {
+      throw PortalBlockchainError.noSupportedCurveForChainId(fromChainId)
+    }
+
+    self.curve = curve
+    self.isMainnet = reference == PortalBlockchain.mainnetReferences[namespace] ?? "NO_MAPPING_FOUND"
```

Comment:
Do we want "NO_MAPPING_FOUND" as the default value if the chain isn't in the lookup?

Reply:
The condition will only complete its evaluation if a value is provided for the right hand side of the check. That's the value of `?? "NO_MAPPING_FOUND"` here. It forces it to evaluate to `false`.